src/audio/google_rtc_audio_processing_mock.c

```diff
@@ -22,29 +24,74 @@ struct GoogleRtcAudioProcessingState {
 	int16_t *aec_reference;
 };
 
-GoogleRtcAudioProcessingState *GoogleRtcAudioProcessingCreate()
+static void SetFormats(GoogleRtcAudioProcessingState *const state,
+		       capture_sample_rate_hz,
+		       int num_capture_input_channels,
+		       int num_capture_output_channels,
+		       int render_sample_rate_hz,
+		       int num_render_channels)
+{
+	state->num_capture_channels = num_capture_input_channels;
+	state->num_output_channels = num_capture_output_channels;
+	state->num_frames = capture_sample_rate_hz / GOOGLE_RTC_AUDIO_PROCESSING_FRAMES_PER_SECOND;
+
+	state->num_aec_reference_channels = num_render_channels;
+	rfree(state->aec_reference);
+	state->aec_reference = rballoc(0,
+				       SOF_MEM_CAPS_RAM,
+				       sizeof(state->aec_reference[0]) *
+				       state->num_frames *
+				       state->num_aec_reference_channels);
```

Comment:
would `rbrealloc()` work for you?

Reply:
@lyakh given this is test code im inclined to reject this change. If it was in the adapter then I would agree